### PR TITLE
fix: fixes for typescript 4.8

### PR DESCRIPTION
**Pull request by joshuaellis**
### Why

- Typescript 4.8 broke some of our types due to narrowing & generics

### What

- Fixed `infer State` using `extend Lookup`
- Used `!` in tests because _we know_ the value will **not** be `null`
- Typecast the `asyncTo` property (possibly not best practice)

### Checklist

- [ ] Documentation updated
- [ ] Demo added
- [x] Ready to be merged

### CODE REVIEW

**Comment by jlamande** [comment]
Hi
is this fix supposed to work with typescript versions prior to 4.8 ?

not sure but we have troubles using the 9.5.3 :

```
node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:116 - error TS1005: '?' expected.
8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State extends Lookup<any> ? Remap<ControllerUpdate<State> & {
                                                                                                                     ~~~~~~~
node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:124 - error TS1005: ':' expected.
8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State extends Lookup<any> ? Remap<ControllerUpdate<State> & {
                                                                                                                             ~~~~~~
node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:136 - error TS1005: ';' expected.
8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State extends Lookup<any> ? Remap<ControllerUpdate<State> & {
                                                                                                                                         ~
node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:15:4 - error TS1005: '(' expected.
15 }> : never;
      ~
node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:15:11 - error TS1005: ')' expected.
15 }> : never;
...
```

**Comment by joshuaellis** [comment]
@jlamande, This could be a couple of things. AFAIK `infer X extends Y` was introduced in `typescript@4.7`. But also, because this was a huge syntax change, you may have to update your `babel`, `esbuild` etc. version to reflect this. It depends what's parsing your TS.

**Comment by gusmagnago** [comment]
I'm facing the same issue on typescript@4.0.5

`8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State extends Lookup<any> ? Remap<ControllerUpdate<State> & {
                                                                                                                     ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:124 - error TS1005: ':' expected.

8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State extends Lookup<any> ? Remap<ControllerUpdate<State> & {
                                                                                                                             ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:136 - error TS1005: ';' expected.

8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State extends Lookup<any> ? Remap<ControllerUpdate<State> & {
                                                                                                                                         ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:15:4 - error TS1005: '(' expected.

15 }> : never;
      ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:15:11 - error TS1005: ')' expected.

15 }> : never;
             ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:20:223 - error TS1005: '?' expected.

20 export declare function useSpring<Props extends object>(props: Function | (() => (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps), deps?: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                 ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:20:231 - error TS1005: ':' expected.

20 export declare function useSpring<Props extends object>(props: Function | (() => (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps), deps?: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                         ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:20:243 - error TS1144: '{' or ';' expected.

20 export declare function useSpring<Props extends object>(props: Function | (() => (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps), deps?: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                     ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:20:265 - error TS1109: Expression expected.

20 export declare function useSpring<Props extends object>(props: Function | (() => (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps), deps?: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                                           ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:20:283 - error TS1005: '(' expected.

20 export declare function useSpring<Props extends object>(props: Function | (() => (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps), deps?: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                                                             ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:20:285 - error TS1005: ';' expected.

20 export declare function useSpring<Props extends object>(props: Function | (() => (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps), deps?: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                                                               ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:28:203 - error TS1005: '?' expected.

28 export declare function useSpring<Props extends object>(props: (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps, deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                             ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:28:211 - error TS1005: ':' expected.

28 export declare function useSpring<Props extends object>(props: (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps, deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                     ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:28:223 - error TS1144: '{' or ';' expected.

28 export declare function useSpring<Props extends object>(props: (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps, deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                 ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:28:245 - error TS1109: Expression expected.

28 export declare function useSpring<Props extends object>(props: (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps, deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                       ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:28:263 - error TS1005: '(' expected.

28 export declare function useSpring<Props extends object>(props: (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps, deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                                         ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:28:265 - error TS1005: ';' expected.

28 export declare function useSpring<Props extends object>(props: (Props & Valid<Props, UseSpringProps<Props>>) | UseSpringProps, deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>, SpringRef<State>] : never;
                                                                                                                                                                                                                                                                           ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:15:193 - error TS1005: '?' expected.

15 export declare function useSprings<Props extends UseSpringProps>(length: number, props: (i: number, ctrl: Controller) => Props, deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                   ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:15:201 - error TS1005: ':' expected.

15 export declare function useSprings<Props extends UseSpringProps>(length: number, props: (i: number, ctrl: Controller) => Props, deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                           ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:15:213 - error TS1144: '{' or ';' expected.

15 export declare function useSprings<Props extends UseSpringProps>(length: number, props: (i: number, ctrl: Controller) => Props, deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                       ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:15:259 - error TS1005: '(' expected.

15 export declare function useSprings<Props extends UseSpringProps>(length: number, props: (i: number, ctrl: Controller) => Props, deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                     ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:15:261 - error TS1005: ';' expected.

15 export declare function useSprings<Props extends UseSpringProps>(length: number, props: (i: number, ctrl: Controller) => Props, deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                       ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:23:215 - error TS1005: '?' expected.

23 export declare function useSprings<Props extends UseSpringsProps>(length: number, props: Props[] & UseSpringsProps<PickAnimated<Props>>[], deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                         ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:23:223 - error TS1005: ':' expected.

23 export declare function useSprings<Props extends UseSpringsProps>(length: number, props: Props[] & UseSpringsProps<PickAnimated<Props>>[], deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                                 ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:23:235 - error TS1144: '{' or ';' expected.

23 export declare function useSprings<Props extends UseSpringsProps>(length: number, props: Props[] & UseSpringsProps<PickAnimated<Props>>[], deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                                             ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:23:281 - error TS1005: '(' expected.

23 export declare function useSprings<Props extends UseSpringsProps>(length: number, props: Props[] & UseSpringsProps<PickAnimated<Props>>[], deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                           ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useSprings.d.ts:23:283 - error TS1005: ';' expected.

23 export declare function useSprings<Props extends UseSpringsProps>(length: number, props: Props[] & UseSpringsProps<PickAnimated<Props>>[], deps: readonly any[] | undefined): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                             ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:8:238 - error TS1005: '?' expected.

8 export declare function useTrail<Props extends object>(length: number, props: (i: number, ctrl: Controller) => UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                               ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:8:246 - error TS1005: ':' expected.

8 export declare function useTrail<Props extends object>(length: number, props: (i: number, ctrl: Controller) => UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                                       ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:8:258 - error TS1144: '{' or ';' expected.

8 export declare function useTrail<Props extends object>(length: number, props: (i: number, ctrl: Controller) => UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                                                   ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:8:300 - error TS1005: '(' expected.

8 export declare function useTrail<Props extends object>(length: number, props: (i: number, ctrl: Controller) => UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                                                                                             ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:8:302 - error TS1005: ';' expected.

8 export declare function useTrail<Props extends object>(length: number, props: (i: number, ctrl: Controller) => UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps?: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                                                                                               ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:10:204 - error TS1005: '?' expected.

10 export declare function useTrail<Props extends object>(length: number, props: UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                              ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:10:212 - error TS1005: ':' expected.

10 export declare function useTrail<Props extends object>(length: number, props: UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                      ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:10:224 - error TS1144: '{' or ';' expected.

10 export declare function useTrail<Props extends object>(length: number, props: UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                  ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:10:266 - error TS1005: '(' expected.

10 export declare function useTrail<Props extends object>(length: number, props: UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                                                            ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTrail.d.ts:10:268 - error TS1005: ';' expected.

10 export declare function useTrail<Props extends object>(length: number, props: UseTrailProps | (Props & Valid<Props, UseTrailProps<Props>>), deps: readonly any[]): PickAnimated<Props> extends infer State extends Lookup<any> ? [SpringValues<State>[], SpringRef<State>] : never;
                                                                                                                                                                                                                                                                              ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:5:235 - error TS1005: '?' expected.

5 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: () => UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps?: any[]): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, PickAnimated<Props>>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                            ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:5:243 - error TS1005: ':' expected.

5 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: () => UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps?: any[]): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, PickAnimated<Props>>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                    ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:5:250 - error TS1144: '{' or ';' expected.

5 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: () => UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps?: any[]): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, PickAnimated<Props>>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                           ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:5:292 - error TS1109: Expression expected.

5 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: () => UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps?: any[]): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, PickAnimated<Props>>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                                     ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:5:314 - error TS1005: '(' expected.

5 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: () => UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps?: any[]): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, PickAnimated<Props>>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                                                           ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:5:316 - error TS1005: ';' expected.

5 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: () => UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps?: any[]): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, PickAnimated<Props>>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                                                             ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:7:240 - error TS1005: '?' expected.

7 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps: any[] | undefined): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, State>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                 ~~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:7:248 - error TS1005: ':' expected.

7 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps: any[] | undefined): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, State>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                         ~~~~~~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:7:255 - error TS1144: '{' or ';' expected.

7 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps: any[] | undefined): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, State>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:7:283 - error TS1109: Expression expected.

7 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps: any[] | undefined): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, State>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                            ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:7:305 - error TS1005: '(' expected.

7 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps: any[] | undefined): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, State>, SpringRefType<State>] : never;
                                                                                                                                                                                                                                                                                                                  ~

node_modules/@react-spring/core/dist/declarations/src/hooks/useTransition.d.ts:7:307 - error TS1005: ';' expected.

7 export declare function useTransition<Item, Props extends object>(data: OneOrMore<Item>, props: UseTransitionProps<Item> | (Props & Valid<Props, UseTransitionProps<Item>>), deps: any[] | undefined): PickAnimated<Props> extends infer State extends Lookup ? [TransitionFn<Item, State>, SpringRefType<State>] : never;
`

**Comment by joshuaellis** [comment]
@gusmagnago see comment above yours.

**Comment by gusmagnago** [comment]
@joshuaellis it wasn't the best for the project I'm working with, but worked, so thank you!  :)

File: packages/types/util.d.ts [hunk 1]
```diff
@@ -118,15 +118,13 @@ type Intersect<U> = (U extends any ? (k: U) => void : never) extends (
   ? I
   : never
 
-export type Exclusive<T> = keyof T extends infer Keys
-  ? Keys extends infer Key
+export type Exclusive<T> = AllKeys<T> extends infer K
```
Comment (chrisnojima) [inline packages/types/util.d.ts:121]: @joshuaellis what is `AllKeys`? My project fails as this doesn't exist?

Comment (joshuaellis) [reply packages/types/util.d.ts:121]: That's odd. I don't understand why `tsc` test would pass if `AllKeys` if undefined.

Comment (joshuaellis) [reply packages/types/util.d.ts:121]: new version 9.5.4 released, 9.5.3 has been deprecated.

**Pull request merged**